### PR TITLE
feat: improve gif classification

### PR DESCRIPTION
**Pull request by YegorZaremba**
Hi @GantMan I working on https://github.com/nsfw-filter/nsfw-filter and have some bugs with `DOM`

### Current state

1) `nsfwjs.min.js` size is 1694185 bytes => ~1.7mb
2) during prediction gif changed size
3) for 10 secs gif it takes 20264ms => ~20 secs

old2

### New

1) `nsfwjs.min.js` size is 2014035 bytes => ~2mb
2) during prediction gif didn't change size
3) for 10 secs gif it takes 15913ms => ~16 secs
4) you can use `fps` property to adjust amount of frames to predict `model.classifyGif(GIF_IMAGE, { topk: 3, fps: 1 })`

new

### CODE REVIEW

**Comment by YegorZaremba** [comment]
UPD1 Oh, I estimate time with prediction of 190 frames for both cases
Found old records, maybe its only gif preparation, without frames prediction

libifjs
23.667 -> 31.377
44.173 -> 52.451
01.013 -> 08.798

gif-frames
47.742 -> 48.779
53.875 -> 54.825
58.736 -> 59.696

**Comment by GantMan** [comment]
This looks great!

Can you please update the README.md to explain the new params along with this change?   Then I can merge.

File: yarn.lock [hunk 1]
```diff
@@ -553,6 +553,26 @@
...
+"@nsfw-filter/gif-frames@1.0.2":
+  version "1.0.2"
+  resolved "https://registry.yarnpkg.com/@nsfw-filter/gif-frames/-/gif-frames-1.0.2.tgz#a00731e56a944c7cfc8e65f26f6f2a6945432ea6"
+  integrity sha512-XZrbJWEN8YfVla5i+PD4Wj51rRlJ8OgnXiPjjOt/OsrbsCR9GZRD4jr953oNWcwiRaoIcOCFWQNMQukO7Yb1dA==
+  dependencies:
+    "@nsfw-filter/save-pixels" "^2.3.4"
+    get-pixels-frame-info-update "3.3.2"
+    multi-integer-range "3.0.0"
+
+"@nsfw-filter/save-pixels@^2.3.4":
```
Comment (YegorZaremba) [inline yarn.lock:565]: I just changed a little bit this lib, https://github.com/nsfw-filter/save-pixels

File: yarn.lock [hunk 2]
```diff
@@ -553,6 +553,26 @@
     "@types/yargs" "^15.0.0"
     chalk "^3.0.0"
 
+"@nsfw-filter/gif-frames@1.0.2":
```
Comment (YegorZaremba) [inline yarn.lock:556]: no changes(except types), just replace with new `@nsfw-filter/save-pixels`

**Comment by YegorZaremba** [comment]
@GantMan Hi, Im finished
- [x] works for both environnements: `browser` and `nodejs`
- [x] tests for `classifyGif`
- [x] updated README docs

ps could you please take a look into `classifyGIF` implementation, maybe I miss sth (sorry, it's pretty messy)

**Comment by GantMan** [comment]
#### FANTASTIC WORK!

**Quick note:**  If I merge this it will break NSFWJS.com for GIFs which is powered by CI to https://github.com/infinitered/nsfwjs/tree/master/example/nsfw_demo

I'm OK merging and then tasking myself with updating the code in the demo, but I guess I should ask.  Do you want to update the small bit of demo code?  That way when I merge this, the new GIF code will go directly to NSFWJS.com?

this means a small update to https://github.com/infinitered/nsfwjs/blob/master/example/nsfw_demo/src/App.js#L123-L127

Again, feel free to refuse.  NSFWJS.com can be broken for a bit until I find time to update the site to the new syntax API.

**Comment by GantMan** [comment]
@YegorZaremba - can you contact me and give me your address?  We'd love to mail you some swag for your contributions.

My email is Gant followed by the company's domain.

**Pull request merged**